src/buf.c

```diff
@@ -51,6 +51,7 @@ struct Buf defs_buf;		/**< for #define's autogenerated. List of strings. */
 struct Buf yydmap_buf;		/**< string buffer to hold yydmap elements */
 struct Buf m4defs_buf;          /**< m4 definitions. List of strings. */
 struct Buf top_buf;             /**< contains %top code. String buffer. */
+struct Buf alloc_buf;           /**< List of strings that need to be freed. */
```

Comment:
Some `Buf`s wrap arrays of int's or char's.  However, others wrap arrays of `char*`s.  The _size_ of the array is what is realloc()ed. But the `char*` is just an entry in the array. Some are assigned from a const value and some from a strdup().  We cannot free all entries, because that would fault since some are const.
(Continued below)